core/src/main/java/com/datastax/oss/driver/internal/core/cql/Conversions.java

```diff
@@ -441,4 +450,77 @@ public static CoordinatorException toThrowable(
         return new ProtocolError(node, "Unknown error code: " + errorMessage.code);
     }
   }
+
+  /** Returns a common size for all kinds of Request implementations. */
+  static int minimumRequestSize(Request statement, DriverContext context) {
+    Preconditions.checkArgument(
+        context instanceof InternalDriverContext,
+        "DriverContext provided cannot be used to calculate statement's size");
+
+    /* Header and payload are common inside a Frame at the protocol level */
+
+    // Frame header has a fixed size of 9 for protocol version > V3, which includes Frame flags size
+    int size = headerEncodedSize();
+
+    if (!statement.getCustomPayload().isEmpty()
+        && ((InternalDriverContext) context)
+            .protocolVersionRegistry()
+            .supports(context.protocolVersion(), ProtocolFeature.CUSTOM_PAYLOAD)) {
+      size += PrimitiveSizes.sizeOfBytesMap(statement.getCustomPayload());
+    }
+
+    /* These are options in the protocol inside a frame that are common to all Statements */
+
+    size += queryFlagsSize(context.protocolVersion().getCode());
+
+    size += PrimitiveSizes.SHORT; // size of consistency level
+    size += PrimitiveSizes.SHORT; // size of serial consistency level
+    size += PrimitiveSizes.LONG; // size of default timestamp
```

Comment:
It's possible if using a server timestamp generator and timestamp is not explicitly set that we won't be sending a timestamp with the request.   I noticed that the 3.x driver also assumes timestamp is always present.  Do you think we should evaluate the statement and timestamp generator from `DriverContext` to determine if the timestamp will be provided?  On the other hand, using server side timestamps is less common these days so maybe not worth it.